File: ProjetAppliSportJPA/src/main/java/AppliSport/model/Utilisateur.java
package AppliSport.model;

import java.util.ArrayList;
import java.util.List;
import java.util.Set;

import javax.persistence.AttributeOverride;
import javax.persistence.AttributeOverrides;
import javax.persistence.Column;
import javax.persistence.Embedded;
import javax.persistence.Entity;
import javax.persistence.ForeignKey;
import javax.persistence.IdClass;
import javax.persistence.JoinColumn;
import javax.persistence.OneToMany;
import javax.persistence.OneToOne;
import javax.persistence.Table;
import javax.persistence.Transient;
import javax.persistence.UniqueConstraint;
import javax.persistence.Version;
import javax.validation.Valid;
import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.Pattern;

@Entity
@Table(name = "utilisateur", uniqueConstraints = { @UniqueConstraint(columnNames = "compte_identifiant", name = "utilisateur_identifiant_uk"), @UniqueConstraint(columnNames = "compte_mail", name = "utilisateur_mail_uk") })
public class Utilisateur extends Compte {
	
	@Valid
	@Embedded
	@AttributeOverrides({ @AttributeOverride(name = "numero", column = @Column(name = "client_numero", length = 50)),
			@AttributeOverride(name = "rue", column = @Column(name = "client_rue", length = 200)),
			@AttributeOverride(name = "codePostal", column = @Column(name = "client_code_postal", length = 20)),
			@AttributeOverride(name = "ville", column = @Column(name = "client_ville", length = 100)) })
	private Adresse adresse;
	
	@NotEmpty
	@Pattern(regexp = "^[a-zA-Z]((-|')?([a-zA-Z]{1,}))*$")
	@Column(name = "utilisateur_nom", length = 100)
	private String nom;
	
	@Pattern(regexp = "^[a-zA-Z]((-|')?([a-zA-Z]{1,}))*$")
	@NotEmpty
	@Column(name = "utilisateur_prenom", length = 100)
	private String prenom;
	
	@NotEmpty
	@Pattern(regexp="^(0|\\+33)[1-9]([-. ]?[0-9]{2}){4}$")
	@Column(name = "utilisateur_tel", length = 30)
	private String numTel;
	
	@OneToMany(mappedBy = "utilisateur")
	private Set<Interet> interets;
	@OneToOne
	@JoinColumn(name = "utilisateur_profil", foreignKey = @ForeignKey(name = "utilisateur_profil_fk"))
	private Profil profilUtilisateur;
	
	@OneToMany(mappedBy = "id.equipe")	
	private List<EquipeUtilisateur> equipes;
	@OneToMany(mappedBy = "id.evenement")
	private List<EvenementUtilisateur> evenements;
	@OneToMany(mappedBy = "id.club")
	private List<ClubUtilisateur> clubs;
	

	public Utilisateur() {
		
	}


	

	public Adresse getAdresse() {
		return adresse;
	}


	public void setAdresse(Adresse adresse) {
		this.adresse = adresse;
	}


	public String getNumTel() {
		return numTel;
	}


	public void setNumTel(String numTel) {
		this.numTel = numTel;
	}


	

	public Set<Interet> getInterets() {
		return interets;
	}




	public void setInterets(Set<Interet> interets) {
		this.interets = interets;
	}




	public List<EquipeUtilisateur> getEquipes() {
		return equipes;
	}




	public void setEquipes(List<EquipeUtilisateur> equipes) {
		this.equipes = equipes;
	}




	public List<EvenementUtilisateur> getEvenements() {
		return evenements;
	}




	public void setEvenements(List<EvenementUtilisateur> evenements) {
		this.evenements = evenements;
	}




	public Profil getProfilUtilisateur() {
		return profilUtilisateur;
	}


	public void setProfilUtilisateur(Profil profilUtilisateur) {
		this.profilUtilisateur = profilUtilisateur;
	}




	public String getNom() {
		return nom;
	}




	public void setNom(String nom) {
		this.nom = nom;
	}




	public String getPrenom() {
		return prenom;
	}




	public void setPrenom(String prenom) {
		this.prenom = prenom;
	}




	public List<ClubUtilisateur> getClubs() {
		return clubs;
	}




	public void setClubs(List<ClubUtilisateur> clubs) {
		this.clubs = clubs;
	}


	
	
	
	

}

File: SportifyApp/src/app/model/utilisateur.ts
import { Adresse } from './adresse';
import { ClubUtilisateur } from './club-utilisateur';
import { Compte } from './compte';
import { EquipeUtilisateur } from './equipe-utilisateur';
import { EvenementUtilisateur } from './evenement-utilisateur';
import { Interet } from './interet';
import { Profil } from './profil';

export class Utilisateur extends Compte {
  private _adresse: Adresse | undefined;
  private _nom: string | undefined;
  private _prenom: string | undefined;
  private _numTel: string | undefined;
  private _profilUtilisateur: Profil | undefined;
  private _interets: Interet[] | undefined;
  private _equipes: EquipeUtilisateur[] | undefined;
  private _evenements: EvenementUtilisateur[] | undefined;
  private _clubs: ClubUtilisateur[] | undefined;

  constructor(
    adresse?: Adresse,
    nom?: string,
    prenom?: string,
    numTel?: string,
    profilUtilisateur?: Profil,
    interets?: Interet[],
    equipes?: EquipeUtilisateur[],
    evenements?: EvenementUtilisateur[],
    clubs?: ClubUtilisateur[]
  ) {
    super();
    this._adresse = adresse;
    this._nom = nom;
    this._prenom = prenom;
    this._numTel = numTel;
    this._profilUtilisateur = profilUtilisateur;
    this._interets = interets;
    this._equipes = equipes;
    this._evenements = evenements;
    this._clubs = clubs;
  }

  /**
   * Getter adresse
   * @return {Adresse }
   */
  public get adresse(): Adresse | undefined {
    return this._adresse;
  }

  /**
   * Getter nom
   * @return {string }
   */
  public get nom(): string | undefined {
    return this._nom;
  }

  /**
   * Getter prenom
   * @return {string }
   */
  public get prenom(): string | undefined {
    return this._prenom;
  }

  /**
   * Getter numTel
   * @return {string }
   */
  public get numTel(): string | undefined {
    return this._numTel;
  }

  /**
   * Getter profilUtilisateur
   * @return {Profil }
   */
  public get profilUtilisateur(): Profil | undefined {
    return this._profilUtilisateur;
  }

  /**
   * Getter interets
   * @return {Interet[] }
   */
  public get interets(): Interet[] | undefined {
    return this._interets;
  }

  /**
   * Getter equipes
   * @return {EquipeUtilisateur[] }
   */
  public get equipes(): EquipeUtilisateur[] | undefined {
    return this._equipes;
  }

  /**
   * Getter evenements
   * @return {EvenementUtilisateur[] }
   */
  public get evenements(): EvenementUtilisateur[] | undefined {
    return this._evenements;
  }

  /**
   * Getter clubs
   * @return {ClubUtilisateur[] }
   */
  public get clubs(): ClubUtilisateur[] | undefined {
    return this._clubs;
  }

  /**
   * Setter adresse
   * @param {Adresse } value
   */
  public set adresse(value: Adresse | undefined) {
    this._adresse = value;
  }

  /**
   * Setter nom
   * @param {string } value
   */
  public set nom(value: string | undefined) {
    this._nom = value;
  }

  /**
   * Setter prenom
   * @param {string } value
   */
  public set prenom(value: string | undefined) {
    this._prenom = value;
  }

  /**
   * Setter numTel
   * @param {string } value
   */
  public set numTel(value: string | undefined) {
    this._numTel = value;
  }

  /**
   * Setter profilUtilisateur
   * @param {Profil } value
   */
  public set profilUtilisateur(value: Profil | undefined) {
    this._profilUtilisateur = value;
  }

  /**
   * Setter interets
   * @param {Interet[] } value
   */
  public set interets(value: Interet[] | undefined) {
    this._interets = value;
  }

  /**
   * Setter equipes
   * @param {EquipeUtilisateur[] } value
   */
  public set equipes(value: EquipeUtilisateur[] | undefined) {
    this._equipes = value;
  }

  /**
   * Setter evenements
   * @param {EvenementUtilisateur[] } value
   */
  public set evenements(value: EvenementUtilisateur[] | undefined) {
    this._evenements = value;
  }

  /**
   * Setter clubs
   * @param {ClubUtilisateur[] } value
   */
  public set clubs(value: ClubUtilisateur[] | undefined) {
    this._clubs = value;
  }
}
